ocaml/xapi/xapi_mgmt_iface.ml

```diff
@@ -29,43 +29,17 @@ let management_interface_server = ref []
 let listening_all = ref false
 let listening_localhost = ref false
 let listening_himn = ref false
-let stunnel_accept = ref None
 let management_m = Mutex.create ()
 
 let update_mh_info interface =
   let (_: string*string) = Forkhelpers.execute_command_get_output !Xapi_globs.update_mh_info_script [ interface ] in
   ()
 
-let stunnel_m = Mutex.create ()
-
-let restart_stunnel_nomutex ~__context ~accept =
+let restart_stunnel ~__context ~accept =
   info "Restarting stunnel (accepting connections on %s)" accept;
-  let xapissl_args = [ "restart"
-                     ; accept
-                     ; "permfile=" ^ !Xapi_globs.server_cert_path
-                     ; Xcp_const.good_ciphersuites
-                     ]
-  in
-  let (_ : Thread.t) = Thread.create (fun () ->
-      Mutex.execute management_m (fun () ->
```

Comment:
This `management_m` mutex is interesting...

I see that `run` takes the mutex. This calls `start`, which calls `restart_stunnel`, which calls `restart_stunnel_nomutex`, which gets us here and blocks. Luckily it's in a thread, so `start` and `run` can complete to unlock the mutex, so that stunnel can be restarted. I'm not sure this worked as intended... Unless there is a very good reason to postpone restarting stunnel until after `run` completes, I think that we're better off without the mutex (as in your patch)... :)

Reply:
i postulate that putting the call to the xapissl script was a workaround for this mutex being locked overzealously

i believe we can remove the thread (in my initial patch, i did not wrap the call to `Helpers.Stunnel.restart` in a thread and tests were looking good)